packages/yoastseo/spec/specHelpers/scoring/seoAssessorTests.js

```diff
@@ -0,0 +1,411 @@
+import Paper from "../../../src/values/Paper";
+import getResults from "../getAssessorResults";
+import buildTree from "../parse/buildTree";
+import EnglishResearcher from "../../../src/languageProcessing/languages/en/Researcher";
+
+/* eslint-disable complexity */
+/**
+ * Checks which assessment are available for an SEO assessor, given a certain Paper.
+ * @param {Assessor} assessor The SEO assessor.
+ * @param {boolean} isProductAssessor Whether this assessor is used for products.
+ * @returns {void}
+ */
+export function checkAssessmentAvailability( assessor, isProductAssessor = false ) {
+	/**
+	 * Finds all assessments that have valid results for a given Paper.
+	 * @param {Paper} paper The paper.
+	 * @returns {string[]} The assessments with valid results.
+	 */
+	function assess( paper ) {
+		const researcher = new EnglishResearcher( paper );
+		buildTree( paper, researcher );
+
+		assessor.assess( paper );
+		return getResults( assessor.getValidResults() );
+	}
+
+	const isStoreBlog = assessor.type.startsWith( "storeBlog" );
+	const isCollection = assessor.type.startsWith( "collection" );
+	const isProduct = assessor.type.startsWith( "product" );
+	const isTaxonomy = assessor.type.startsWith( "taxonomy" );
+
+	const defaultAssessments = [ "keyphraseLength", "metaDescriptionLength", "titleWidth", "textLength" ];
+	if ( isStoreBlog ) {
+		defaultAssessments.pop();
+	}
+
+	let requiresTextAssessments = [ "images", "externalLinks", "internalLinks" ];
+	if ( isCollection || isStoreBlog || isTaxonomy ) {
+		requiresTextAssessments = [];
+	}
+	if ( isProduct ) {
+		requiresTextAssessments = [ "images" ];
+	}
+
+	const requiresKeyphraseAssessments = [ "introductionKeyword" ];
+	if ( isStoreBlog ) {
+		requiresKeyphraseAssessments.pop();
+	}
```

Comment:
This part triggers an ESLint warning for complexity. Not sure if there's an opportunity here to simplify the code, however.